### ISSUE TITLE
one-to-one relationships

### ISSUE THREAD

**Issue by Trikke** [pardom-zz/ActiveAndroid#107]
Hi, just a quick question.

Is this supported somehow? Or is another database model better in a situation where Class1 can only hold one reference to Class2?

Thanks!

**Comment by leonardoxh** [comment]
Yep. If you put a object of Class2 in Class1 and annotate the column will work (one-to-one) :+1:

**Comment by vsigler** [comment]
Well, you can just define two models that reference each other. No big deal. Or is there a special requirement there somewhere? Like you want the reference both ways?

``` java
@Table(name="ModelOne")
public class ModelOne extends Model {
   @Column(name="fkModelTwo")
   private ModelTwo modelTwo;

   public void setModelTwo(ModelTwo modTwo) {
       this.modelTwo = modTwo;
   }

   public ModelTwo getModelTwo() {
      return this.modelTwo;
   }
}

@Table(name="ModelTwo")
public class ModelTwo extends Model {
   @Column(name="fkModelOne")
   private ModelOne modelOne;

   public void setModelOne(ModelOne modOne) {
       this.modelOne = modOne;
   }

   public ModelOne getModelOne() {
      return this.modelOne;
   }
}
```

Then you'd just save them like this:

``` java
ModelOne m1 = new ModelOne();
m1.save();

ModelTwo m2 = new ModelTwo();
m2.setModelOne(m1);

m2.save();
m1.setModelTwo(m2);
m1.save();
```

**Comment by Trikke** [comment]
Thanks for the quick response, that's awesome!

Maybe, for completeness, you should add that example to the relationship part of the docs?

**Comment by Trikke** [comment]
Hmm, I just tried this setup, and when using "loadFromCursor" on "ModelOne" ( i get my results from a cursor ), it seems to generate an stackoverflow. probably because it is bouncing around between those 2 models.

java.lang.StackOverflowError
        at java.util.WeakHashMap.poll(WeakHashMap.java:550)
        at java.util.WeakHashMap.put(WeakHashMap.java:589)
        at android.database.sqlite.SQLiteConnectionPool.finishAcquireConnectionLocked(SQLiteConnectionPool.java:911)
        at android.database.sqlite.SQLiteConnectionPool.tryAcquirePrimaryConnectionLocked(SQLiteConnectionPool.java:847)
        at android.database.sqlite.SQLiteConnectionPool.waitForConnection(SQLiteConnectionPool.java:613)
        at android.database.sqlite.SQLiteConnectionPool.acquireConnection(SQLiteConnectionPool.java:348)
        at android.database.sqlite.SQLiteSession.acquireConnection(SQLiteSession.java:894)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:586)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
        at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:44)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1314)
        at android.database.sqlite.SQLiteDatabase.rawQuery(SQLiteDatabase.java:1253)
        at com.activeandroid.util.SQLiteUtils.rawQuery(SQLiteUtils.java:92)
        at com.activeandroid.util.SQLiteUtils.rawQuerySingle(SQLiteUtils.java:100)
        at com.activeandroid.query.From.executeSingle(From.java:216)
        at com.activeandroid.Model.loadFromCursor(Model.java:234)
        at com.activeandroid.util.SQLiteUtils.processCursor(SQLiteUtils.java:206)
        at com.activeandroid.util.SQLiteUtils.rawQuery(SQLiteUtils.java:93)
        at com.activeandroid.util.SQLiteUtils.rawQuerySingle(SQLiteUtils.java:100)
        at com.activeandroid.query.From.executeSingle(From.java:216)
        at com.activeandroid.Model.loadFromCursor(Model.java:234)
        at com.activeandroid.util.SQLiteUtils.processCursor(SQLiteUtils.java:206)
        at com.activeandroid.util.SQLiteUtils.rawQuery(SQLiteUtils.java:93)
        at com.activeandroid.util.SQLiteUtils.rawQuerySingle(SQLiteUtils.java:100)
        at com.activeandroid.query.From.executeSingle(From.java:216)
        at com.activeandroid.Model.loadFromCursor(Model.java:234)
        at com.activeandroid.util.SQLiteUtils.processCursor(SQLiteUtils.java:206)
        at com.activeandroid.util.SQLiteUtils.rawQuery(SQLiteUtils.java:93)
        at com.activeandroid.util.SQLiteUtils.rawQuerySingle(SQLiteUtils.java:100)
        at com.activeandroid.query.From.executeSingle(From.java:216)
        at com.activeandroid.Model.loadFromCursor(Model.java:234)

**Comment by vsigler** [comment]
Hmm, I never tried the double-linked setup like that, it was just an idea. You're right, it recursively tries to load the referenced entity. As there is no code to handle posible circular references, it results in stack overflow.

There would need to be either code to handle circular references or support for lazy-loading, so the linked model would be loaded upon access.

**Comment by Trikke** [comment]
Should i create a new bug report on this? It seems like an issue and something that someone can and probably will recreate.

**Comment by Defuera** [comment]
Well this was very useful, I think you really need to add this to the tutorial. I spend quite a while trying to figure out how this should work. 

Ps. thanks for a wonderful library!

**Comment by marcohc** [comment]
Is this issue still open? any solution? thanks!

**Comment by ghost** [comment]
I face a problem be caused by this issue. Does anyone have solution? thanks!

**Comment by jlhonora** [comment]
What if you store only the other model's id? A bit ugly but gets the job done.

**Comment by ixsans** [comment]
is this issue has any solution? I got same problem.

**Comment by marcohc** [comment]
@ixsans it seems this library is dead :S look at the latest commits...
